packages/world/src/systems/MineSystem.sol

```diff
@@ -109,7 +109,17 @@ contract MineSystem is System {
       minedType = RandomResourceLib._collapseRandomOre(mined, coord);
     }
 
-    uint128 finalMass = MassReductionLib._processMassReduction(caller, callerCoord, mined, toolSlot);
+    (uint128 finalMass, uint128 energyReduction) = _processEnergyReduction(caller, callerCoord, mined, toolSlot);
+
+    {
+      if (energyReduction > 0) {
+        // If player died, return early
+        (uint128 callerEnergy,) = transferEnergyToPool(caller, energyReduction);
+        if (callerEnergy == 0) {
+          return mined;
```

Comment:
hmm this will consume tool mass but not set the terrain mass right?

Reply:
ah you are right, but it is kind of a chicken and egg problem 🤔  because we need to use the tool to see if we need to consume from player